content/docs/command-reference/commit.md

```diff
@@ -17,65 +17,52 @@ positional arguments:
-
-- Sometimes we want to edit source code, config, or data files in a way that
-  doesn't cause changes in the results of their data pipeline. We might write
-  add code comments, change indentation, remove some debugging printouts, or any
-  other change that doesn't cause changed stage outputs. However, DVC will
-  notice that some <abbr>dependencies</abbr> have changed, and expect you to
-  reproduce the whole pipeline. If you're sure no pipeline results would change,
-  use `dvc commit` to force update the `dvc.lock` or `.dvc` files and cache.
-
-- In some cases, we have previously executed a stage, and later notice that some
-  of the files/directories used by the stage as dependencies or created as
-  outputs are missing from `dvc.yaml`. It is possible to
-  [add missing data to an existing stage](/docs/user-guide/how-to/add-deps-or-outs-to-a-stage),
-  and then `dvc commit` can be used to save outputs to the cache (and update
-  `dvc.lock`)
-
-- It's always possible to manually execute the command or source code used in a
-  stage without DVC (outputs must be unprotected or removed first in certain
-  cases, see `dvc unprotect`). Once the desired result is reached, use
-  `dvc commit` to update the `dvc.lock` file(s) and store changed data to the
-  cache.
-
-Let's take a look at what is happening in the first scenario closely. Normally
-DVC commands like `dvc add`, `dvc repro` or `dvc run` commit the data to the
-<abbr>cache</abbr> after creating or updating a `dvc.lock` or `.dvc` file. What
-_commit_ means is that DVC:
-
-- Computes a hash for the file/directory.
-- Enters the hash value and file name in the `dvc.lock` or `.dvc` file.
-- Tells Git to ignore the file/directory (adding them to `.gitignore`). (Note
-  that if the <abbr>project</abbr> was initialized with no Git support
-  (`dvc init --no-scm`), this does not happen.)
-- Adds the file(s) in question to the cache.
-
-There are many cases where the last step is not desirable (for example rapid
-iterations on an experiment). The `--no-commit` option prevents it (on the
-commands where it's available). The file hash is still computed and added to the
-`dvc.lock` or `.dvc` file, but the actual data is not cached. And this is where
-the `dvc commit` command comes into play: It performs that last step when
-needed.
-
-Note that it's best to avoid the last three scenarios. They essentially
-force-update the `dvc.lock` or `.dvc` files and save data to cache. They are
-still useful, but keep in mind that DVC can't guarantee reproducibility in those
-cases.
+This command can be useful for several scenarios, when
+[stages](/doc/command-reference/run) are under development, or to force-update
+the cache without having to execute stage commands again (avoiding `dvc repro`).
+
+> Note that this does not register project versions. That can be done with
+> `git commit` (if using Git).
+
+Normally, `dvc repro` and `dvc run` finish up with the same steps as `dvc add`,
```

Comment:
> Are there any significant important fixes/changes here?

@shcheklein yes. The whole structure changed to emphasize the main use (development process i.e. a companion for add/run/repro --no-commit) in the first half. The 2nd half of the desc. goes over the other scenarios where this might be useful.

All the explanations were simplified and implementation details were concentrated in a single paragraph or 2.

Reply:
it's a very large change ... what was a specific problem that we are solving? can we do a fix instead of rewriting the whole doc?

Reply:
There were 2 problems: (1) to focus on the main use case of `commit` (putting advanced uses lower in the description); and (2) fix the details about what commit does. I rewrote (2) since it had errors (the part about "Normally, `dvc repro` and `dvc run` finish up with the same steps as `dvc add`"), the remaining changes consist of moving paragraphs the bottom of the description (1).